src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -171,6 +182,79 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     })()
   }, [assetsById, chainAdapter, dispatch, wallet])
 
+  const txIds = useSelector(selectTxIds)
+  const txsById = useSelector(selectTxs)
+  const txHistoryStatus = useSelector(selectTxHistoryStatus)
+
+  /**
+   * monitor for new pending txs, add them to a set, so we can monitor when they're confirmed
+   */
+  useEffect(() => {
+    // we only want to refetch portfolio if a new tx comes in after we're finished loading
+    if (txHistoryStatus !== 'loaded') return
```

Comment:
This will probably create an edge case in which if a tx has been started - and not confirmed yet, as the user opens up the app, this effect returns here because `txHistoryStatus` is not `loaded` yet, but we lose track of pending txs and we can't determine when they're confirmed, so portfolio balance will be incorrect.

Reply:
yup, totally aware of this too, i'm ok with this being an uncovered edge case